src/ReTestItems.jl

```diff
@@ -640,6 +640,20 @@ function is_testsetup_file(filepath)
     )
 end
 
+# like `relpath` but assumes `path` is nested under `startdir`, else just returns `path`
+function nestedrelpath(path, startdir)
+    path == startdir && return "."
+    relp = chopprefix(path, startdir)
+    sep = Base.Filesystem.path_separator
+    if endswith(startdir, sep)
+        return relp
+    elseif startswith(relp, sep)
+        return chopprefix(relp, sep)
+    else
+        return path
+    end
+end
```

Comment:
```suggestion
function nestedrelpath(path, startdir)
    path == startdir && return SubString(".", 1:1)
    relp = chopprefix(path, startdir)
    sep = Base.Filesystem.path_separator
    if endswith(startdir, sep)
        return relp
    elseif startswith(relp, sep)
        return chopprefix(relp, sep)
    else
        return @view(path[1:end])
    end
end
```

With this suggestion, the function becomes type stable for `String` inputs (always returns a `SubString{String}`) and I get 0 allocs:

```julia
julia> @btime nestedrelpath($path, $("test"))
  54.048 ns (0 allocations: 0 bytes)
"dir/foo_test.jl"
```

Reply:
thanks <3